File: src/java/LeetCode/maxproduct_subarray/MaxProductSubArray_Naive.java
package LeetCode.maxproduct_subarray;

public class MaxProductSubArray_Naive {
    public static int maxProductNaive(int[] nums) {
        int n = nums.length;
        if (n == 0) return 0;
        int maxProduct = Integer.MIN_VALUE;
        for (int i = 0; i < n; i++) {
            long prod = 1;
            for (int j = i; j < n; j++) {
                prod *= nums[j];
                if (prod > maxProduct) maxProduct = (int) prod;
            }
        }
        return maxProduct;
    }

    public static void main(String args[]){
        int nums[] = new int[]{1,-2,-3,4};
        System.out.println("Max product subarray is : "+maxProductNaive(nums));
    }
}

File: src/python/LeetCode150/maxproduct_subarray/maxproduct_subarray_naive.py
def maxProduct_naive(nums):
    n = len(nums)
    if n == 0:
        return 0
    max_product = float('-inf')
    for i in range(n):
        prod = 1
        for j in range(i, n):
            prod *= nums[j]
            if prod > max_product:
                max_product = prod
    return int(max_product)


# Example usage
if __name__ == "__main__":
    # Input: List of numbers
    nums = [7, 1, -5, 3, 6, -4]
    print("Maximum product sub array:", maxProduct_naive(nums))